doc/spec_exceptionHandling.dox

```diff
@@ -69,12 +71,18 @@ namespace ChimeraTK {
+  - \anchor b_3_1 3.1 After successfully re-opening the device, a recovery procedure is executed before allowing any read/write operations from the ApplicationModules and FanOuts again. This recovery procedure involves the following steps:
+    - \anchor exceptionHandling_b_3_1_0 3.1.0 In a recovery group, devices wait until all involved devices successfully complete the open step before starting the initialisation handler in B.3.1.1 ("barrier OPEN"). [link to test]
+    - \anchor exceptionHandling_b_3_1_1 3.1.1 The so-called initialisation handlers are executed (see \ref exceptionHandling_b_3_2 "3.2") [\ref testExceptionHandling_b_3_1_1 "T"].
+      - \anchor exceptionHandling_b_3_1_1_1 3.1.1.1 The device is closed before the initialisation handler is called, and reopened afterwards. This allows backends which can only be opened once to use external init handler scripts. \ref exceptionHandling_comment_b_3_1_1_1 "(*)" [link to test]
+      - \anchor exceptionHandling_b_3_1_1_2 3.1.1.2 In a recovery group, devices wait until all involved devices successfully complete the init handler step before restoring register values in B.3.1.2 ("barrier INIT_HANDLER"). [link to test]
+    - \anchor exceptionHandling_b_3_1_2 3.1.2 All registers that have been written since the start of the application are restored with their latest values. The register values are restored in the same order they were written. Registers of the type ChimeraTK::Void are not written. \ref comment_b_3_1_2 "(*)" [\ref testExceptionHandling_b_3_1_2 "T"]
+    - \anchor exceptionHandling_b_3_1_2_1 3.1.2.1 In a recovery group, devices wait until all involved devices successfully completed the register value restoring before activating the synchronous read in B.3.1.3 ("barrier WRITE_RECOVERY"). [link to test]
     - \anchor exceptionHandling_b_3_1_3 3.1.3 The asynchronous read transfers of the device are (re-)activated by calling Device::activateAsyncReads(). [\ref testExceptionHandling_b_3_1_3 "T"]
     - \anchor exceptionHandling_b_3_1_4 3.1.4 Finally, \c Devices/\<alias\>/deviceBecameFunctional is written to inform any module subscribing to this variable about the finished recovery. \ref comment_b_3_1_4 "(*)" [\ref testExceptionHandling_b_3_1_4 "T"]
   - \anchor exceptionHandling_b_3_2 3.2 Any number of initialisation handlers can be added to the DeviceModule in the user code. Initialisation handlers are callback functions which will be executed when a device is opened for the first time and after a device recovers from an exception, before any application-initiated transfers are executed (including delayed write transfers). See DeviceModule::addInitialisationHandler(). [\ref testExceptionHandling_b_3_2 "T"]
+  - \anchor exceptionHandling_b_3_3 3.3 The application must teminate cleanly, even if the recovery is waiting at one of the barriers mentioned in \ref b_3_1 "3.1" [link to test]
```

Comment:
```suggestion
  - \anchor exceptionHandling_b_3_3 3.3 The application terminates cleanly, even if the recovery is waiting at one of the barriers mentioned in \ref b_3_1 "3.1" [link to test]
```
(otherwise it is a requirement to the Application...)